testsuite/tests/trmc/usage_warnings.ml

```diff
@@ -0,0 +1,115 @@
+(* TEST
+   * expect *)
+
+(* build-up *)
+let[@tailrec_mod_constr] rec append xs ys =
+  match xs with
+  | [] -> ys
+  | x :: xs -> x :: append xs ys
+[%%expect {|
+val append : 'a list -> 'a list -> 'a list = <fun>
+|}]
+
+(* incorrect version: this cannot work *)
+let[@tailrec_mod_constr] rec flatten = function
+  | [] -> []
+  | xs :: xss -> append xs (flatten xss)
+[%%expect {|
+Line 3, characters 17-40:
+3 |   | xs :: xss -> append xs (flatten xss)
+                     ^^^^^^^^^^^^^^^^^^^^^^^
+Warning 70: This call is in tail-position in a TRMC function, but
+the function called is not itself specialized for TRMC,
+so the call will not be in tail position in the transformed
+version.
+Please either mark the called function with the
+[@tailrec_mod_constr] attribute, or mark this call with the
+[@tailcall false] attribute to make its non-tailness explicit.
+Lines 1-3, characters 39-40:
```

Comment:
Actually I think that seeing this warning is an artifact of using toplevel-style tests (with the current implementation, if the whole file was processed at once there would be no warning).

The *actual* warning we want to see is that this [@tailrec_mod_constr] definition is useless and wrong since the call to `flatten` itself is not TRMC-optimized. But in fact the current implementation does not provide this warning (in the case where the call to `append` itself is correctly TRMC-ed), as it only expects calls in TRMC position.

I don't know whether I want to have a warning on non-TRMC-position self-and-mutual calls that are not marked `[@tailcall false]` (this sounds potentially useful in some cases, including this one, but also annoying in other cases and also annoying to implement), or accept that we don't give a very useful warning in this example.